### PR TITLE
feat: add "unhandledrejection" event support

### LINKED ISSUE #7013: onunhandledrejection for globalThis?
I was looking for an equivalent to the Node.js "unhandledrejection" event in Deno, but it appears there isn't one on `globalThis`.

There does appear to be an `onunhandledrejection` event handler [defined on workers](https://github.com/denoland/deno/blob/master/cli/dts/lib.webworker.d.ts#L5468), though.

Is there an equivalent event outside of workers, and if not, could there be? :smile:

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit adds support for "unhandledrejection" event.

This event will trigger event listeners registered using:
- "globalThis.addEventListener("unhandledrejection")
- "globalThis.onunhandledrejection"

This is done by registering a default handler using
"Deno.core.setPromiseRejectCallback" that allows to
handle rejected promises in JavaScript instead of Rust.

This commit will make it possible to polyfill 
"process.on("unhandledRejection")" in the Node compat
layer.

Closes #7013

### CODE REVIEW

**Review by bnoordhuis — APPROVED** [review]
Thanks for doing this, Bartek! The test failures are probably (hopefully) a small matter of updating test expectations.

**Comment by bartlomieju** [comment]
> Thanks for doing this, Bartek! The test failures are probably (hopefully) a small matter of updating test expectations.

I think there's quite more to do before this PR will be ready to merge. Just wanted to get ball rolling for now.

**Comment by bartlomieju** [comment]
Discussed with @bnoordhuis offline, we're punting this PR for now as it's not clear how this functionality would integrate with Node compatibility layer. Ben will first implement `process.on("unhandledRejection", ...)` in `deno_std/node`.

File: ext/web/02_event.js [hunk 1]
```diff
@@ -1224,6 +1224,58 @@
...
+        keys: [
+          ...EVENT_PROPS,
+          "promise",
+          "reason",
+        ],
+      }));
+    }
+
+    // TODO(lucacasonato): remove when this interface is spec aligned
+    [SymbolToStringTag] = "PromiseRejectionEvent";
```
[the remark this answers is not in the document]
Comment (lucacasonato) [reply ext/web/02_event.js:1271]: The entire eventing system has not been migrated to use our webidl helper classes yet. It is on my backburner.

**Comment by bartlomieju** [comment]
> Does this PR not implement `document.addEventListener("unhandledrejection")`?

It does implement it.

**Comment by Skillz4Killz** [comment]
As a reminder, it is incredibly useful to some people to be able to make sure this can be used to prevent a process from crashing in production.

**Comment by hashemisa** [comment]
What version should we wait for this possibility?

**Comment by jiawei397** [comment]
Do you have any plans to merge this submission?

**Comment by bartlomieju** [comment]
> Do you have any plans to merge this submission?

Yes, but it hasn't been the highest priority.

**Comment by jiawei397** [comment]
> > Do you have any plans to merge this submission?
> 
> Yes, but it hasn't been the highest priority.

OK, I will continue to pay attention to this. Thanks for your reply.

File: core/bindings.rs [hunk 2]
```diff
@@ -340,7 +340,7 @@ pub extern "C" fn promise_reject_callback(message: v8::PromiseRejectMessage) {
   } else {
     let promise = message.get_promise();
     let promise_global = v8::Global::new(scope, promise);
-
+    eprintln!("we are here");
```
Comment (cjihrig) [inline core/bindings.rs:343]: 🙈

**Review by kitsonk — APPROVED** [review]
LGTM

**Comment by bartlomieju** [comment]
Note to self: add test for when promise rejects with "undefined" to check that "reason" on the event is actually "undefined" instead of "null".

**Review by lucacasonato — APPROVED** [review]
LGTM!

**Pull request merged**